### PR TITLE
Overwrite module titles with ones defined in collection.xml

### LINKED ISSUE #365: Module titles set in collection.xml overwriten by original titles from index.cnxml
# Scenario

When modules are included in a collection under a different name than they're published under, that new name should be used in table of contents, but currently is overwritten with the original.

# Reproducibility Details

- **Impact area**: book which include modules under different name than they were published under
- **User**: N/A
- **URL / Location**: N/A
- **How many times were you able to reproduce the issue?** every time

## Steps to reproduce:

- Step 1 Create a collection with includes a module under a different name
- Step 2 Convert collection to XHTML
- Step 3 Note that, in the raw XHTML, while modules will have the original title in content, the title in metadata will be the one set by collection
- Step 4 Bake collection
- Step 5 Note that now titles in module metadata and in TOC are overwritten with original titles.

# Environment:

- **Instance**: N/A
- **OS**: any
- **Browser**: N/A

# Impact

- **Severity**: ?
- **Frequency of occurrence**: all books

### PR DESCRIPTION
**Pull request by aiwenar**
The new mixin, `modify_syncTitlesFromCollection`, replaces titles within modules' content with ones from modules' metadata. Passes are moved one number up because it needs to run before `modify_spanWrapTitles` (otherwise module's title will include both the overridden and the original title).

Fixes #365

### CODE REVIEW

File: recipes/mixins/_modify.scss [hunk 1]
```diff
@@ -61,6 +61,21 @@
...
+
+// Replace module titles with the ones set by collection.
+@mixin modify_syncTitlesFromCollection() {
+  div[data-type="page"] {
+    > div[data-type="metadata"] > [data-type="document-title"] {
+      string-set: page-title content();
+    }
+
+    > [data-type="document-title"] {
+      content: string(page-title);
```
Comment (reedstrm) [inline recipes/mixins/_modify.scss]: This depends on `page -> document-title` always coming after `page->metadata->document-title` in the doc, and the metadata-stanza one being the override title. I'm pretty sure our transformation process does in fact guarantee  that, but we never state it precisely anywhere. I *can* promise that the `page -> document-title` is the native page title, and any `metadata -> document-title` is the title-from-book, but the  fact that they always fall in this order is almost accidental. A useful accident, however. :-P

Comment (reedstrm) [reply recipes/mixins/_modify.scss]: Thinking more about the existing code, I think we can depend on that order staying. The only thing we _do_ need is to write some tests for this one.

**Comment by philschatz** [comment]
@aiwenar In order to add a test, you can do the following:

1. create a snippet file in `recipes/mixins/styleguide/book.page.title.snippet.xml` containing something like the following XML
1. create a comment in `recipes/mixins/styleguide/_all.scss` that refers to the file above (but the file extension will be different)
1. run `./script/generate-guide statistics` (or just `./script/test`) and see if the Page title was overridden

```xml
<div xmlns="http://www.w3.org/1999/xhtml" data-type="chapter">
  <div data-type="page">
    <div data-type="document-title">Incorrect Title</div>
    <div data-type="metadata">
      <div data-type="title">Correctly Overridden Title that should appear in the H2 element</div>
    </div>

    <para xmlns="http://cnx.rice.edu/cnxml">This is the contents of the section (aka Page)</para>
    <para xmlns="http://cnx.rice.edu/cnxml">It can contain multiple paragraphs, notes, exercises, etc</para>
  </div>
</div>
```

**Comment by philschatz** [comment]
This PR has **expected** differences in the Index (tested with `./script/diff-book macroecon`)

```sh
git fetch origin pull/366/merge:katalyst-366
git checkout katalyst-366
./script/compile-books 
time ./script/diff-book --all
diff ./data/macroecon-prepared.xhtml ./data/macroecon-baked.xhtml
```


```diff
498c498
<                   >What Is Economics, and Why Is It Important?</span
---
>                   >What Economics Is and Why It's Important</span
5871c5871
<             >What Is Economics, and Why Is It Important?</span
---
>             >What Economics Is and Why It's Important</span
9386c9386
<               >What Is Economics, and Why Is It Important?</span
---
>               >What Economics Is and Why It's Important</span
199527c199527
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
200717c200717
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
201021c201021
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
201695c201695
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
201889c201889
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
201903c201903
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
207511c207511
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
207765c207765
<               >1.1 What Is Economics, and Why Is It Important?</span
---
>               >1.1 What Economics Is and Why It's Important</span
```

**Comment by philschatz** [comment]
(continuation from previous comment because of max size limitations)

# TEA HS Physics

This also caused additional elements to show up (like AP Biology)

<details>

<summary>Click me to see the diff</summary>

```diff
$ diff ./data/TEAhsphysics-prepared.xhtml ./data/TEAhsphysics-baked.xhtml
3361a3362,3369
>       <div
>             class="description"
>             data-type="description"
>             itemprop="description"
>           >
>         
>         <div
>           />
3362a3371,3372
>       </div
>         >
4414a4425,4434
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
6039a6060,6069
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
7062a7093,7102
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
26071a26112,26121
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
28407a28458,28467
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
32108a32169,32178
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
34857a34928,34937
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
44929a45010,45019
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
48363a48454,48463
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
61164a61265,61274
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
61849a61960,61969
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
63018a63139,63148
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
67359a67490,67499
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
76921a77062,77071
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
81351a81502,81511
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
93799a93960,93969
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
105074a105245,105254
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
113323a113504,113513
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
131180a131371,131380
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
136278a136479,136488
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
140928a141139,141148
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
154305a154526,154535
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
159443a159674,159683
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
170999a171240,171249
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
176889a177140,177149
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
179054a179315,179324
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
198823a199094,199103
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
201887a202168,202177
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
204596a204887,204896
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
214869a215170,215179
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
216940a217251,217260
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
230459a230780,230789
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
233869a234200,234209
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
237875a238216,238225
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
248513a248864,248873
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
249145a249506,249515
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
253342a253713,253722
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
256028a256409,256418
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
268387a268778,268787
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
269636a270037,270046
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
273050a273461,273470
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
279729a280150,280159
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
282491a282922,282931
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
287257a287698,287707
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
291013a291464,291473
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
306580a307041,307050
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
308930a309401,309410
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
318347a318828,318837
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
322630a323121,323130
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
329170a329671,329680
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
340105c340615
<                       >30</span
---
>                       >31</span
340207c340717
<                       >31</span
---
>                       >32</span
340789a341300,341309
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
345506a346027,346036
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
354946a355477,355486
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
360360a360901,360910
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
367009a367560,367569
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
370374a370935,370944
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
380216a380787,380796
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
396568a397149,397158
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
402554a403145,403154
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
410263c410863
<                 >4</span
---
>                 >5</span
410344c410944
<                 >5</span
---
>                 >6</span
410544a411145,411154
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
423018c423628
<                 >6</span
---
>                 >7</span
423059c423669
<                 >7</span
---
>                 >8</span
423249a423860,423869
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
429572c430192
<                 >8</span
---
>                 >9</span
429614c430234
<                 >9</span
---
>                 >10</span
436591c437211
<                       >29</span
---
>                       >30</span
436627c437247
<                       >30</span
---
>                       >31</span
436663c437283
<                       >31</span
---
>                       >32</span
436705c437325
<                       >32</span
---
>                       >33</span
436781c437401
<                       >33</span
---
>                       >34</span
436821c437441
<                       >34</span
---
>                       >35</span
436861c437481
<                       >35</span
---
>                       >36</span
437043c437663
<                       >36</span
---
>                       >37</span
437093c437713
<                       >37</span
---
>                       >38</span
437192c437812
<                       >38</span
---
>                       >39</span
437255c437875
<                       >39</span
---
>                       >41</span
437292c437912
<                       >40</span
---
>                       >42</span
437794a438415,438424
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
444618a445249,445258
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
451105a451746,451755
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
458128c458778
<                       >8</span
---
>                       >9</span
458165c458815
<                       >9</span
---
>                       >10</span
458240c458890
<                       >10</span
---
>                       >11</span
458276c458926
<                       >11</span
---
>                       >12</span
458312c458962
<                       >12</span
---
>                       >13</span
458374c459024
<                       >13</span
---
>                       >14</span
458414c459064
<                       >14</span
---
>                       >15</span
458448c459098
<                       >15</span
---
>                       >16</span
458503c459153
<                       >16</span
---
>                       >17</span
458543c459193
<                       >17</span
---
>                       >18</span
458673c459323
<                       >18</span
---
>                       >19</span
458738c459388
<                       >19</span
---
>                       >21</span
458780c459430
<                       >20</span
---
>                       >22</span
458866c459516
<                       >21</span
---
>                       >23</span
461905a462556,462565
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
465109a465770,465779
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
467953a468624,468633
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
472261c472941
<                       >9</span
---
>                       >10</span
472334c473014
<                       >10</span
---
>                       >11</span
472374c473054
<                       >11</span
---
>                       >12</span
472414c473094
<                       >12</span
---
>                       >13</span
472454c473134
<                       >13</span
---
>                       >14</span
472516c473196
<                       >14</span
---
>                       >15</span
472556c473236
<                       >15</span
---
>                       >16</span
472598c473278
<                       >16</span
---
>                       >17</span
472640c473320
<                       >17</span
---
>                       >18</span
472708c473388
<                       >18</span
---
>                       >19</span
472748c473428
<                       >19</span
---
>                       >20</span
472790c473470
<                       >20</span
---
>                       >21</span
472826c473506
<                       >21</span
---
>                       >22</span
472893c473573
<                       >22</span
---
>                       >23</span
472944c473624
<                       >23</span
---
>                       >24</span
473016c473696
<                       >24</span
---
>                       >25</span
473066c473746
<                       >25</span
---
>                       >26</span
473138c473818
<                       >26</span
---
>                       >27</span
473188c473868
<                       >27</span
---
>                       >28</span
473270c473950
<                       >28</span
---
>                       >29</span
474698c475378
<                       >26</span
---
>                       >27</span
474740c475420
<                       >27</span
---
>                       >28</span
474780c475460
<                       >28</span
---
>                       >29</span
474830c475510
<                       >29</span
---
>                       >30</span
474880c475560
<                       >30</span
---
>                       >31</span
475085c475765
<                       >31</span
---
>                       >32</span
475125c475805
<                       >32</span
---
>                       >33</span
475187c475867
<                       >33</span
---
>                       >34</span
475227c475907
<                       >34</span
---
>                       >35</span
475267c475947
<                       >35</span
---
>                       >36</span
475317c475997
<                       >36</span
---
>                       >37</span
475367c476047
<                       >37</span
---
>                       >38</span
475409c476089
<                       >38</span
---
>                       >39</span
475449c476129
<                       >39</span
---
>                       >40</span
475489c476169
<                       >40</span
---
>                       >41</span
475563c476243
<                       >41</span
---
>                       >42</span
475613c476293
<                       >42</span
---
>                       >43</span
475672c476352
<                       >43</span
---
>                       >44</span
475722c476402
<                       >44</span
---
>                       >45</span
475784c476464
<                       >45</span
---
>                       >46</span
475824c476504
<                       >46</span
---
>                       >47</span
475864c476544
<                       >47</span
---
>                       >48</span
475914c476594
<                       >48</span
---
>                       >49</span
475972c476652
<                       >49</span
---
>                       >50</span
476012c476692
<                       >50</span
---
>                       >51</span
476062c476742
<                       >51</span
---
>                       >52</span
476143c476823
<                       >52</span
---
>                       >53</span
476679a477360,477369
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
486963c487653
<                 >1</span
---
>                 >9</span
487134a487825,487834
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
492614c493314
<                 >2</span
---
>                 >14</span
492785a493486,493495
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
498239a498950,498959
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
502271a502992,503001
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
508620c509350
<                       >1</span
---
>                       >5</span
508660c509390
<                       >2</span
---
>                       >6</span
508723c509453
<                       >3</span
---
>                       >10</span
508795c509525
<                       >4</span
---
>                       >11</span
508835c509565
<                       >5</span
---
>                       >12</span
508875c509605
<                       >6</span
---
>                       >13</span
508915c509645
<                       >7</span
---
>                       >14</span
508951c509681
<                       >8</span
---
>                       >15</span
509023c509753
<                       >9</span
---
>                       >17</span
509063c509793
<                       >10</span
---
>                       >18</span
509124c509854
<                       >11</span
---
>                       >19</span
509166c509896
<                       >12</span
---
>                       >20</span
509232c509962
<                       >13</span
---
>                       >22</span
509300c510030
<                       >14</span
---
>                       >23</span
509361c510091
<                       >15</span
---
>                       >24</span
509401c510131
<                       >16</span
---
>                       >25</span
509441c510171
<                       >17</span
---
>                       >26</span
509518c510248
<                       >18</span
---
>                       >27</span
509769c510499
<                       >1</span
---
>                       >7</span
509830c510560
<                       >2</span
---
>                       >9</span
509874c510604
<                       >3</span
---
>                       >10</span
509946c510676
<                       >4</span
---
>                       >11</span
509998c510728
<                       >5</span
---
>                       >23</span
510064c510794
<                       >6</span
---
>                       >29</span
510122c510852
<                       >7</span
---
>                       >34</span
510188c510918
<                       >8</span
---
>                       >41</span
510252c510982
<                       >9</span
---
>                       >46</span
510330c511060
<                       >10</span
---
>                       >53</span
510500c511230
<                       >11</span
---
>                       >54</span
510563c511293
<                       >12</span
---
>                       >58</span
510626c511356
<                       >13</span
---
>                       >62</span
510688c511418
<                       >14</span
---
>                       >65</span
511247a511978,511987
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
514206a514947,514956
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
519391a520142,520151
>               class="description"
>               data-type="description"
>               itemprop="description"
>             >
>         
>         <div
>             />
>       
>       </div
>           ><div
527288a528049,528058
>             class="description"
>             data-type="description"
>             itemprop="description"
>           >
>         
>         <div
>           />
>       
>       </div
>         ><div
542483a543254,543261
>       <div
>             class="description"
>             data-type="description"
>             itemprop="description"
>           >
>         
>         <div
>           />
542484a543263,543264
>       </div
>         >
```

</details>

**Comment by aiwenar** [comment]
## Title changes
(Macroecon, Physics)

Title for module m48591 is _What Is Economics, and Why Is It Important?_, but it's used in collection as _What Economics Is and Why It's Important_ (according to `index.cnxml` and `collection.xml` from offline zip), so this change is correct. Same is true for _Physics_ (although it's already called “Preface” in [on-line version](https://cnx.org/contents/Ax2o07Ul@9.98:pFeekPiU@17/Preface), so I'm not sure why it shows in the diff?).

Perhaps it's just something that was overlooked during initial migration from legacy rendering to baking-based one?

## Unexpected additions/changes
(AP Biology, TEA AP Biology, TEA AP Physics, TEA HS Physics, TEA HS Statistics)

Any chance you could provide me with complete source and pre-baking XHTML? I'd like to check what's going on.

## Validation error
(TEA Macro/Microecon AP)

I'll wait for Connexions/cnx-easybake#85 with this one.

## Katalyst physics

Yes, they're all correct (that's why we made this PR after all).

**Comment by openstaxalina** [comment]
@philschatz do we need to test the TEA titles?  That content has already been delivered.

**Comment by philschatz** [comment]
@aiwenar I forgot to clarify that the comments I made had 2 audiences: expected changes like those in Macroecon and Physics (so Alina can update the content) and **unexpected** changes like those in AP Biology, TEA AP Biology, TEA AP Physics, TEA HS Physics, TEA HS Statistics. So please ignore the **expected changes** 😊 .

@openstaxalina I think we need to maintain those so that they do not diverge when/if we need to send updates to TEA or when we start baking more books. But maybe we don't need to worry about the TEA titles any more?

**Comment by aiwenar** [comment]
@philschatz I think you have just found an issue with my internal natural language parser: I've noticed “unexpected” in the first sentence and assumed it applies to the entire comment, without realising it's also specified per section 😆

**Comment by philschatz** [comment]
I screwed up too; I should have typed `This PR contains **expected** AND **unexpected** differences` especially since I was trying to target 2 different audiences with one comment 😊  (retroactively fixed after the damage was already done)

**Comment by philschatz** [comment]
@aiwenar have you made any progress on finding these differences? I thought I sent the books but have not heard back on any problems you ran into

**Comment by philschatz** [comment]
@aiwenar reported that he did not find any regressions after running the baked HTML files through `xmllint`. I will go through and try to reproduce

**Comment by philschatz** [comment]
I re-ran the scripts on my machine (without the extra `xmllint` step). Most of the books were fine but I found the following issues:

- `apbiology`: has links that used to be empty now contain Page titles
- `TEAapbiology` and `TEAhsphysics`: have additional metadata descriptions that are added (probably the Book metadata description is added to each Page)
- `apbiology` and `TEAapphysics`: some `<h2>` elements become `<span>` elements
- `TEAhsstats` : some `<h4>` elements become `<span>` elements
- `TEAmacroeconap` and `TEAmicroeconap`: produce **StackTraces** (see below)

```
Traceback (most recent call last):
  File "/.../cnx-recipes/venv/bin/cnx-epub-validate-collated", line 11, in <module>
    load_entry_point('cnx-epub', 'console_scripts', 'cnx-epub-validate-collated')()
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/scripts/collated_single_html/main.py", line 49, in main
    binder = reconstitute(args.collated_html)
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/collation.py", line 51, in reconstitute
    return adapt_single_html(xhtml)
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/adapters.py", line 359, in adapt_single_html
    _adapt_single_html_tree(binder, body[0], nav_tree, top_metadata=metadata)
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/adapters.py", line 472, in _adapt_single_html_tree
    child.xpath('./*[@data-type="metadata"]')[0])
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/html_parsers.py", line 84, in parse_metadata
    return parser()
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/html_parsers.py", line 118, in __call__
    return self.metadata
  File "/.../cnx-recipes/venv/src/cnxepub/cnxepub/html_parsers.py", line 138, in metadata
    "A value for '{}' could not be found.".format(key))
ValueError: A value for 'title' could not be found.
./script/bake-book: ERROR: could not run [cnx-epub-validate-collated ./data/TEAmicroeconap-baked.xhtml]
./script/diff-book: ERROR: could not run [./script/bake-book TEAmicroeconap --use-repeatable-ids]
```

**Comment by openstaxalina** [comment]
cannot duplicate the second expected change.  The module title shows as "Preface" both within and outside of the collection for me: https://cnx.org/contents/pFeekPiU.

**Comment by philschatz** [comment]
Ah yeah, I am not sure why `recipes/books/_TEAspecific.scss` contains passes (rather than putting them in `_generator.scss`). That probably explains most of these regressions. Thanks for noticing & fixing that!

@helenemccarron could the logic in `recipes/books/_TEAspecific.scss` be moved into `_generator.scss`?


- `apbiology` looks good (adds the Page title to the heading/ToC in addition to the number)
- `TEAapbiology` no longer has regressions
- `TEAhsstats` no longer has regressions
- `TEAmacroeconap` no longer produces Stack Traces

I think that except for deciding when/if the "Move logic into `_generator.scss`" comment above should be handled this is ready for QA (& therefore merging). Any objections?

**Pull request merged**